content/lua/concepts/strings/terms/find/find.md

```diff
@@ -0,0 +1,176 @@
+---
+
+The **`.find()`** function is used to search for a substring within a string and returns the start and end indices of the first occurrence of the substring.
+
+If the substring is not found, the function returns `nil`.
+
+## Syntax
+
+These are 2 different syntaxes for the .find() function:
+
+### 1
+
+```pseudo
+string.find(fullString, searchString, init, pattern)
+```
+
+This requires assigning the function to a variable, to return the function.
+
+or
+
+### 2
+
+```pseudo
+startindex, endindex = string.find(fullstring, searchstring, init, pattern)
+```
+
+This uses multiple assignment to save the results to an individual variables.
+
+- `fullString`: Is the string we are searching through.
+- `searchString`: Is the sub-string to search for.
+- `init` (optional): Is the initial index where the search starts. *Default index is 1*.
+- `pattern` (optional): Is a Boolean flag indicating whether to search using **pattern matching**[^1]. When set to false, pattern matching can be used, where as when set to true, pattern matching is ignored, and exact matches are searched for. *Default is false*.
+
+## Example 1
+
+Use `.find()` without multiple assignments:
+
+```lua
+sentence = 'hello world'
+search1 = string.find(sentence, "hello")
+search2 = string.find(sentence, "world")
+search3 = string.find(sentence, "universe")
+
+print(search1)
+print(search2)
+print(search3)
+```
+
+The above example assigns the function to a unique variable, in order for the results to be returned/printed to the console.
+
+The results of the above functions are:
+
+```shell
+1
+7
+nil
+```
+
+- `hello` is found at starting index 1 in the `sentence` string.
+- `world` is found at starting index 7 in the `sentence` string.
+- `universe` is not found in `sentence` string, therefore the output is nil.
+
+## Example 2
+
+Use `.find()` with multiple assignments:
+
+```lua
+sentence = "hello world"
+a, b = string.find(sentence, "hello")
+print(a)
+print(b)
+
+print(a, b)
+
+y, z = string.find(sentence, "world")
+print(y)
+print(z)
+
+print(y, z)
+```
+
+The above example uses multiple assignment, to save the results to individual variables, in order for the results to be returned/printed to the console.
+
+The results of the above functions are:
+
+```shell
+1
+5
+1  5
+7
+11
+7  11
+```
+
+- `hello` starts at index 1, and ends at index 5 in the `sentence` string. Due to multiple assignments, `a` saves the value `1`, and `b` saves the second value `5`.
+- `world` starts at index 7, and ends at index 11 in the `sentence` string. Due to multiple assignments, `y` saves the first value `7`, and `z` saves the second value `11`.
+
+## Example 3
+
+Use `.find()` with `init` (Initial position) parameter:
+
+```lua
+sentence1 = 'hello world'
+search = string.find(sentence1, "l", 5)
+print(search)  -- Output: 10
+
+sentence2 = 'world world'
+a, b = string.find(sentence2, "orl", 5)
+print(a)  -- Output: 8
+print(b)  -- Output: 10
+
+print(a, b)  -- Output: 8  10
```

Comment:
```suggestion
sentence1 = 'hello world'
search = string.find(sentence1, "l", 5)
print(search)  

sentence2 = 'world world'
a, b = string.find(sentence2, "orl", 5)
print(a)  
print(b)

print(a, b)  
```
no need to add comments here

Reply:
yep, this was a mistake. I aimed to remove all comments, but forgot to remove these before the PR.